src/expr/src/linear.rs

```diff
@@ -1412,20 +1419,38 @@ pub mod util {
 
     use crate::MirScalarExpr;
 
-    /// Return the map associating columns in the logical,
-    /// unthinned representation of a collection to columns in the
-    /// thinned representation of the arrangement corresponding to `key`.
+    #[allow(dead_code)]
+    /// A triple of actions that map from rows to (key, val) pairs and back again.
+    struct KeyValRowMapping {
+        /// Expressions to apply to a row to produce key datums.
+        to_key: Vec<MirScalarExpr>,
+        /// Columns to project from a row to produce residual value datums.
+        to_val: Vec<usize>,
+        /// Columns to project from the concatenation of key and value to reconstruct the row.
+        to_row: Vec<usize>,
+    }
```

Comment:
Should we keep this?

Reply:
I think we should, and we should eventually use it more. Right now I think we pass around a lot of `(Vec<MirScalarExpr>, Vec<usize>, Vec<usize>)` and need to remember which fields are which. The structure would allow us to have a documented representation of the information that is harder to misuse.